### ISSUE TITLE
Segmentation fault and integrality constraint violation

### ISSUE THREAD

**Issue by sbronson** [coin-or/Cbc#463]
For the model below, with preprocessing enabled, cbc's solution violates an integrality constraint.  With preprocessing disabled, it segfaults.  Tested on master (7206b34), built with `coinbrew build Cbc --enable-debug`.

```
Minimize
 obj: 0 a

Subject To
 c1: + x2 = 8.7
 c2: + a - x2 = 0
 c3: + b1 + b2 = 1
 c4: - x3 + x4 <= 0
 c5: + x3 - x4 + 10 b1 <= 10
 c6: - x1 + x4 <= 1
 c7: - x4 + 9 b2 <= 0

Bounds
 -Inf <= a <= Inf
 x1 = 8
 0 <= x2 <= 10
 0 <= x3 <= 10
 0 <= x4 <= 9

Generals
 a

Binaries
 b1
 b2

End
```

```
$ cbc t.lp -solve -solution -
Welcome to the CBC MILP Solver
Version: devel
Build Date: Dec 18 2021

[...]

Optimal - objective value 0.00000000
      0 a                    8.7                       0
      1 x2                   8.7                       0
      2 b1                     1                       0
      3 b2                     0                       0
      4 x3                     0                       0
      5 x4                     0                       0
      6 x1                     8                       0
```

It says a=8.7, but a is an integer variable.

Adding `-preprocess off` results in a segmentation fault.

```
$ cbc t.lp -preprocess off -solve -solution -
Option for preprocess changed from sos to off
Continuous objective value is 0 - 0.00 seconds
1 infeasibilities
Problem is infeasible - tightenPrimalBounds!
Segmentation fault
```

gdb
```
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7e85d76 in CbcModel::secondaryStatus (this=0x0) at [...]/CbcModel.hpp:1118
1118        return secondaryStatus_;

#0  0x00007ffff7e85d76 in CbcModel::secondaryStatus (this=0x0) at [...]/CbcModel.hpp:1118
#1  0x00007ffff7f3dabb in CbcMain1 (argc=7, argv=0x7fffffffde28, model=..., callBack=0x55555555540b <dummyCallBack(CbcModel*, int)>, parameterData=...) at [...]/CbcSolver.cpp:10374
#2  0x00005555555554e8 in main (argc=7, argv=0x7fffffffde28) at [...]/CoinSolve.cpp:362
```

**Comment by jjhforrest** [comment]
Thanks,

Presumably something to do with reading .lp format.  Should be easy to find.

John Forrest

**Comment by jjhforrest** [comment]
Apologies - nothing to do with .lp format.

**Comment by jjhforrest** [comment]
Both errors were because the problem is infeasible.  The second error - without preprocessing was found in an unusual place and so one CbcModel was NULL so crash.  The first error was due to an error in preprocessing free integer variables.  My first fix tidied lower bounds to -1.0e14 - which is good enough for most real world problems!  However I have now  put in a cleaner fix.  Should be OK now.
